aws/usageReports/ebs/monthlyReport.go

```diff
@@ -0,0 +1,263 @@
+		} else if cast, ok := err.(*elastic.Error); ok && cast.Details.Type == "search_phase_execution_exception" {
+			l.Error("Error while getting data from ES", map[string]interface{}{
+				"type":  fmt.Sprintf("%T", err),
+				"error": err,
+			})
+		} else {
+			l.Error("Query execution failed", map[string]interface{}{"error": err.Error()})
+		}
+		return nil, errors.GetErrorMessage(ctx, err)
+	}
+	return res, nil
+}
+
+// getSnapshotInfoFromEs gets information about an snapshot from previous report to put it in the new report
+func getSnapshotInfoFromES(ctx context.Context, snapshot utils.CostPerResource, account string, userId int) Snapshot {
+	var docType SnapshotReport
+	var snap = Snapshot{
+		SnapshotBase: SnapshotBase{
+			Id:          snapshot.Resource,
+			Description: "N/A",
+			State:       "N/A",
+			Encrypted:   false,
+			StartTime:   time.Time{},
+		},
+		Tags:  make([]utils.Tag, 0),
+		Cost: 0,
+		Volume: Volume{
+			Id:    "N/A",
+			Size:  -1,
+		},
+	}
+	res, err := getElasticSearchEbsSnapshot(ctx, account, snapshot.Resource,
+		es.Client, es.IndexNameForUserId(userId, IndexPrefixEBSReport))
+	if err == nil && res.Hits.TotalHits > 0 && len(res.Hits.Hits) > 0 {
+		err = json.Unmarshal(*res.Hits.Hits[0].Source, &docType)
+		if err == nil {
+			snap.Id = docType.Snapshot.Id
+			snap.Description = docType.Snapshot.Description
+			snap.State = docType.Snapshot.State
+			snap.Encrypted = docType.Snapshot.Encrypted
+			snap.StartTime = docType.Snapshot.StartTime
+			snap.Tags = docType.Snapshot.Tags
+		}
+	}
+	return snap
+}
+
+// fetchMonthlySnapshotsList sends in snapshotInfoChan the snapshots fetched from DescribeSnapshots
+// and filled by DescribeSnapshots and getSnapshotStats.
+func fetchMonthlySnapshotsList(ctx context.Context, creds *credentials.Credentials, snap utils.CostPerResource,
+	account, region string, snapshotChan chan Snapshot, userId int) error {
+	defer close(snapshotChan)
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	sess := session.Must(session.NewSession(&aws.Config{
+		Credentials: creds,
+		Region:      aws.String(region),
+	}))
+	svc := ec2.New(sess)
+	desc := ec2.DescribeSnapshotsInput{SnapshotIds: []*string{aws.String(getResourceId(snap))}}
+	snapshots, err := svc.DescribeSnapshots(&desc)
+	if err != nil {
+		snapshotChan <- getSnapshotInfoFromES(ctx, snap, account, userId)
+		return err
+	}
+	for _, snapshot := range snapshots.Snapshots {
+		logger.Debug("DATA SNAPSHOT =======", map[string]interface{}{
+			"region": region,
+			"snap": snap,
+			"data": snapshot,
+		})
+		snapshotChan <- Snapshot{
+			SnapshotBase: SnapshotBase{
+				Id:          aws.StringValue(snapshot.SnapshotId),
+				Description: aws.StringValue(snapshot.Description),
+				State:       aws.StringValue(snapshot.State),
+				Encrypted:   aws.BoolValue(snapshot.Encrypted),
+				StartTime:   aws.TimeValue(snapshot.StartTime),
+				Region:      region,
+			},
+			Tags:  getSnapshotTag(snapshot.Tags),
```

Comment:
Make sure to run `gofmt` to have the right alignment for data in these structures

Reply:
Fixed